core/sail/api/src/main/java/org/eclipse/rdf4j/sail/helpers/AbstractSailConnection.java

```diff
@@ -459,6 +459,10 @@ public final void rollback()
 						txnPrepared = false;
 					}
 				}
+				else {
+					logger.warn("Cannot rollback transaction on connection because transaction is not active",
+							new Throwable());
```

Comment:
Hm, yeah, perhaps a good idea. You can just read the private boolean `debugEnabled` by the way, so no need to read the system prop yourself:

`logger.warn("Cannot rollback transaction on connection because transaction is not active", debugEnabled ? new Throwable() : null);`